app/employee-app/backend/modules/database/db_functions.bal

```diff
@@ -78,3 +79,111 @@ public isolated function getEmployees(EmployeeFilter filters, int 'limit, int of
             lead: dbEmployeeInfo.lead == 1
         };
 }
+
+# Retrieve organizational details from HRIS.
+#
+# + employeeStatuses - List of employee statuses to consider
+# + return - List of business units
+public isolated function getOrgStructure(string[]? employeeStatuses) returns OrgDataResponse|error {
+
+    OrgData[] orgDataDBResponse = [];
+    stream<OrgDataDB, sql:Error?> resultStream = databaseClient->query(getOrgStructureQuery(employeeStatuses));
+    error? iterateError = from OrgDataDB orgData in resultStream
+        do {
+            Department[]|error departments = orgData.children.fromJsonStringWithType();
+            if departments is error {
+                string errorMsg = string `An error occurred when retrieving departments data of ${orgData.name}!`;
+                return error(errorMsg, departments);
```

Comment:
```suggestion
                return error(`An error occurred when retrieving departments data of ${
                    orgData.name}!`, departments);
```
No need to define a variable for this since it's not getting re-used

Reply:
ACK, but need to use `string` before the message body.

Reply:
Yes. Better to use string backtick template. 
https://learn-ballerina.github.io/best_practices/string_concat.html?highlight=string%20template#concatenate-string-values

Reply:
Already used backtick template.